test/parallel/test-stream-readable-event.js

```diff
@@ -83,3 +83,29 @@ const Readable = require('stream').Readable;
     r.on('readable', common.mustCall());
   }, 1);
 }
+
+{
+  // pushing a empty string in non-objectMode should
+  // trigger next `read()`.
+  const underlyingData = ['', 'x', 'y', '', 'z'];
+  let result = '';
+  const r = new Readable();
+  r._read = function() {
+    process.nextTick(() => {
+      if (!underlyingData.length) {
+        this.push(null);
+      } else {
+        this.push(underlyingData.shift());
+      }
+    });
+  };
+
+  r.on('readable', () => {
+    const data = r.read();
+    if (data !== null) result += data;
```

Comment:
could you collect the chunks in an array and deepEqual those and the end instead of the concatenated string? That'd make it a bit easier for me (and others) to understand and show that it isn't returning an empty string here ever :)

Reply:
There is a misunderstanding. This change only fixed that the 'end' event isn't emitted when we push a empty string, a empty buffer or `undefined`.

Reply:
Currently, a `ReadableStream` in non-object mode doesn't inform users a empty string or a empty buffer.

Reply:
I'd like it if the test was a bit more explicit.

`result += data` doesn't tell me if `r.read()` returns an empty buffer, which from reading the test, you might expect.

Doing `result.push(data)` and then on end `assert.deepEquals(result, [...])` would help

Reply:
@mafintosh I did it.